builtin/providers/aws/resource_aws_s3_bucket.go

```diff
@@ -538,15 +538,18 @@ func resourceAwsS3BucketRead(d *schema.ResourceData, meta interface{}) error {
 		}
 	}
 
-	//read the acceleration status
-	accelerate, err := s3conn.GetBucketAccelerateConfiguration(&s3.GetBucketAccelerateConfigurationInput{
-		Bucket: aws.String(d.Id()),
-	})
-	log.Printf("[DEBUG] S3 bucket: %s, read Acceleration: %v", d.Id(), accelerate)
-	if err != nil {
-		return err
+	s3HostedRegion := meta.(*AWSClient).region
+	if s3HostedRegion != "cn-north-1" && s3HostedRegion != "us-gov-west-1" {
```

Comment:
@kwilczynski Pretty much, except I would still log full error message. I think we even have a helper function somewhere for matching these AWS errors.